### ISSUE TITLE
Doesn't seem to take into consideration categories

### ISSUE THREAD

**Issue by jverkoey** [mattstevens/objc-diff#8]
In the following before/after situation, objc-diff seems to believe that `someMethod` is deleted rather than deprecated. My guess is that the diff engine may not be considering APIs across categories.

Before:

```
@interface MyController : UIViewController
@end

@interface MyController ()

- (SomeObject *)someMethod;

@end
```

After:

```
@interface MyController : UIViewController

- (SomeObject *)otherMethod;

@end

@interface MyController ()

- (SomeObject *)someMethod
__deprecated_msg("Use otherMethod instead.");

@end
```

**Comment by mattstevens** [comment]
Good catch, it looks like objc-diff is incorrectly ignoring class extensions due to a check that skips anonymous declarations. I'll have a look over the weekend.

**Comment by jverkoey** [comment]
I have some time to look into this right now if there's a good place to start :)

**Comment by jverkoey** [comment]
I'm currently building objectdoc under the initial assumption that perhaps objectdoc was out of date. I'm curious where the "check that skips anonymous declarations" might be found.

**Comment by mattstevens** [comment]
This happens in [`shouldIncludeEntityAtCursor:`](https://github.com/mattstevens/objc-diff/blob/95998186066dc66e50e291605d011d8b6a4d4d52/OCDiff/OCDAPIComparator.m#L203). I believe this issue is entirely on the objc-diff side; ObjectDoc is creating the cursors properly and objc-diff is ignoring these particular cursors.

You can exercise this by adding a test to OCDAPIComparatorTests (a big test suite I should really split up) or creating a simple test header with a class extension and using the `--new` option to see what the tool is parsing.

**Comment by jverkoey** [comment]
Sent a pull request; thanks for pointing in the right direction!

**Comment by mattstevens** [comment]
Fixed via 9600f66b6b9444ae959523c42bb8735c746d5d43 and f522b656724e1b32ab225e6d38c34e11c6aede96.
